modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -271,22 +281,51 @@ object Inputs {
 
     assert(validElems.nonEmpty)
 
-    val (inferredWorkspace, inferredNeedsHash, workspaceOrigin) = validElems
-      .collectFirst {
-        case d: Directory => (d.path, true, WorkspaceOrigin.SourcePaths)
+    val (inferredWorkspace, inferredNeedsHash, workspaceOrigin) = {
+      val settingsFiles = validElems.flatMap {
```

Comment:
Does the `Seq` of `settingsFiles` contain different configuration files? What happened in this case?

```
$ ls src
Hello.scala
project.settings.scala
$ scala-cli src src/project.settings.scala
```